lib/evaluation/evaluationClientWrapper.ts

```diff
@@ -65,25 +66,36 @@ export class EvaluationClientWrapper {
       const marked: HandlerCheckResult[] = this._client.results.filter(
         (result) => result.evaluator === 'MarkForSubmissionEvaluation' && result.status !== Status.ERROR
       );
-      const matchSubmissionRequirements = this.matchWithoutSubmissionRequirements(marked, presentationDefinition);
-      const matches = this.extractMatches(matchSubmissionRequirements);
-      const credentials: IVerifiableCredential[] = matches.map(
-        (e) =>
-          jp.nodes(
-            this._client.wrappedVcs.map((wrapped) => wrapped.original),
-            e
-          )[0].value
-      );
-      selectResults = {
-        errors: errors,
-        matches: [...matchSubmissionRequirements],
-        areRequiredCredentialsPresent: Status.INFO,
-        verifiableCredential: credentials,
-        warnings,
-      };
+      const checkWithoutSRResults: HandlerCheckResult[] = this.checkWithoutSubmissionRequirements(marked, presentationDefinition);
+      if (!checkWithoutSRResults.length) {
+        const matchSubmissionRequirements = this.matchWithoutSubmissionRequirements(marked, presentationDefinition);
+        const matches = this.extractMatches(matchSubmissionRequirements);
+        const credentials: IVerifiableCredential[] = matches.map(
+          (e) =>
+            jp.nodes(
+              this._client.wrappedVcs.map((wrapped) => wrapped.original),
+              e
+            )[0].value
+        );
+        selectResults = {
+          errors: [], // fixme: should this be empty?
```

Comment:
New fixme introduced

Reply:
reminder

Reply:
removed the fixme, we have to fix this and similar log handling during next round of refactors